rustls-libssl/src/entry.rs

```diff
@@ -1579,6 +1579,18 @@ entry! {
     }
 }
 
+entry! {
+    pub fn _SSL_SESSION_set1_id_context(
+        sess: *mut SSL_SESSION,
+        sid_ctx: *const c_uchar,
+        sid_ctx_len: c_uint,
+    ) -> c_int {
+        let slice = try_slice!(sid_ctx, sid_ctx_len);
+        try_clone_arc!(sess).get_mut().set_context(slice);
```

Comment:
The upstream docs note the `sid_ctx_len` must be less than [`SSL_MAX_SID_CTX_LENGTH`](https://github.com/openssl/openssl/blob/067fbc01b9e867b31c71091d62f0f9012dc9e41a/include/openssl/ssl.h.in#L63)

Should we uphold that invariant in the event we implement `SSL_SESSION_get0_id_context` and don't want to break a downstream assumption?

Reply:
Thanks!